tests/cpp/test_gpu_tensorcore.cpp

```diff
@@ -1671,175 +1671,215 @@ TEST_F(GPUTTensorCoreTest, FusionAmpereViewMatmulTN_CUDA) {
+    gemm_tile.warp_tile = GemmTile(64, 64, 32);
+    gemm_tile.instruction_tile = GemmTile(16, 8, 16);
 
-  tv1cw->setMemoryType(MemoryType::Shared);
-  // Schedule mma input
-  tv0cr->applyMmaSwizzle(MmaOperand::A);
+    // [M,K]
+    auto tv0 = makeContigTensor(2, DataType::Half);
+    // [N,K] or [K, N] if use_mkn_dim_order is true
+    auto tv1 = makeContigTensor(2, DataType::Half);
+    // Propagate the allocation domain post-broadcast.
+    if (use_mkn_dim_order) {
+      tv1->setAllocationDomain({tv1->axis(1), tv1->axis(0)}, true);
+    }
+    fusion.addInput(tv0);
+    fusion.addInput(tv1);
 
-  // [... Mi, Ni, Ki]
-  tv0b->reorder({{-2, -3}, {-3, -2}});
-  tv0b->applyMmaSwizzle(MmaOperand::A);
+    // [M,N,K] or [M, K, N] if use_mkn_dim_order is true
+    auto tv0b = use_mkn_dim_order ? broadcast(tv0, {false, false, true})
+                                  : broadcast(tv0, {false, true, false});
+    auto tv1b = broadcast(tv1, {true, false, false});
 
-  tv1cr->applyMmaSwizzle(MmaOperand::B);
-  tv1b->applyMmaSwizzle(MmaOperand::B);
+    // propagate the allocation domain in case there's one.
+    if (use_mkn_dim_order) {
+      tv1b->setAllocationDomain(
+          {tv1b->axis(0), tv1b->axis(2), tv1b->axis(1)}, true);
+    }
 
-  // Schedule mma output
-  tv2c->applyMmaSwizzle(MmaOperand::Accumulator);
-  tv2->applyMmaSwizzle(MmaOperand::Accumulator);
+    // [M, K, N] or [M, N, K]
+    auto tv2 = use_mkn_dim_order ? fusedMultiplySum(tv0b, tv1b, {1})
+                                 : fusedMultiplySum(tv0b, tv1b, {2});
 
-  // Parallelize
-  //  0   1  2  3   4   5  6   7  8  9  10
-  // [Mo No Ko Kwo Mwo Nwo Mw Nw (Mi Ni Ki)]
-  tv2c->axis(4)->parallelize(ParallelType::TIDz);
-  tv2c->axis(5)->parallelize(ParallelType::TIDy);
+    fusion.addOutput(tv2);
 
-  // Parallelize
-  //  0  1  2   3   4   5  6  7
-  // [Mo No Mwo Nwo Mw Nw (Mi Ni)]
-  tv2->axis(0)->parallelize(ParallelType::BIDx);
-  tv2->axis(1)->parallelize(ParallelType::BIDy);
-  tv2->axis(2)->parallelize(ParallelType::TIDz);
-  tv2->axis(3)->parallelize(ParallelType::TIDy);
+    auto mma_ops = ir_utils::getOpsOfType<MmaOp>(&fusion);
+    NVF_CHECK(
+        1 == mma_ops.size(),
+        "Invalid number of MmaOp instances in fusion definition, expected 1, got ",
+        mma_ops.size());
+    mma_ops.front()->setMacro(MmaMacro::Turing_16_8_16);
+
+    auto tv0cw = tv0->cacheAfter(LoadStoreOpType::CpAsync);
+    auto tv0cr = tv0cw->cacheAfter(LoadStoreOpType::LdMatrix);
+    auto tv1cw = tv1->cacheAfter(LoadStoreOpType::CpAsync);
+    auto tv1cr = tv1cw->cacheAfter(LoadStoreOpType::LdMatrix);
+
+    // In the case that input B has an (transposed [M, N, K] ) allocation domain
+    // and it's rfactor domain in of the order [M, K, N], reorder it to the
+    // expected [M, N, K] and propagate it backwards.
+    if (use_mkn_dim_order) {
+      tv2->reorder({{1, 2}, {2, 1}});
+      scheduler_utils::BoundedDirectionalTransformPropagator::backward(
```

Comment:
> We create a permutation vector based on the allocation domain for tv1b and use it to reorder the output of the mma op. The changes to the output are then propagated back up the fusion.

I just realized that this is done explicitly in this test.

I think this is where @jacobhinkle was asking. i.e. What does `backward` propagation is supposed to do? are we expecting this backward propagation to change any allocation domain and have we verified that it's still the same test?

My question here is, should we validate the propagation? I'm happy with it as-is.

Reply:
I think this question still holds, is this propagation supposed to change anything and should we validate that?

Reply:
NVM, I think Prontonu just mentioned this is used to propagate the re-order. So I guess the scheduling following afterwards is already validating the correctness.

Reply:
Yes, the propagation should change the order of the leaf nodes from the output of the mma op all the way to the inputs of the fusion. I can add checks for that.